### PR TITLE
Simulate integer index fix

### LINKED ISSUE #8487: BUG: time index from simulate method not correct for integer index not starting from zero
#### Describe the bug

The prediction index in the forecasting models simulation output is not correct for the special case where you have a range or integer index that doesn't start from zero. The  resulting index is as if the endog index started at 0. 

For example, if the index of the endog is [3, 4, 5, 6, 7, 8] the expected index for nsimulations=3 is [9, 10, 11] but what we get instead is [7, 8, 9].

I discovered this while writing an adapter for the statsmodels simulate method in sktime https://github.com/sktime/sktime/pull/3462 cause some of the automated sktime output index checks were failing.

#### Code Sample, a copy-pastable example if possible

```python
from statsmodels.tsa.statespace.dynamic_factor import DynamicFactor # same with SARIMAX and ETS
import pandas as pd
import numpy as np

index = pd.Int64Index(np.arange(12) + 3)

y = pd.DataFrame(
    data= {
        0: np.arange(12),
        1: np.arange(12)/2 + 5,
    },
    index=index
)

forecaster = DynamicFactor(endog=y, k_factors=1, factor_order=1)
forecaster = forecaster.fit()

simulations = forecaster.simulate(
    nsimulations=5,
    repetitions=3,
    anchor="end"
)

expected_simulation_index = pd.Int64Index(np.arange(np.max(index)+1, np.max(index)+6))

assert np.all(simulations.index ==  expected_simulation_index)
```
<details>
This issue is persistent across all the forecasting algorithms I've tried. 
</details>

#### Expected Output
The index when `anchor = "end"` starts from one time step after the endog max index.

#### Output of ``import statsmodels.api as sm; sm.show_versions()``

<details>

Using the current version from the main branch.
Originally spotted it on version 0.12.1. 

</details>

### PR DESCRIPTION
**Pull request by ltsaprounis**
- [X] closes #8487
- [ ] tests added / passed. 
- [X] code/documentation is well formatted.  
- [X] properly formatted commit message. See 
      [NumPy's guide](https://docs.scipy.org/doc/numpy-1.15.1/dev/gitwash/development_workflow.html#writing-the-commit-message). 

<details>

**Notes**:

The PR is fixing the issue and not breaking any tests in statsmodels/tsa/statespace/tests/test_simulate.py. 
I will also add a test for this edge case before making this ready for review.

</details>

### CODE REVIEW

**Comment by ltsaprounis** [comment]
@ChadFulton - Thanks for the feedback! I changed the code based on your recommendation in https://github.com/statsmodels/statsmodels/issues/8487 and added a warning and a test.

File: statsmodels/tsa/statespace/tests/test_simulate.py [hunk 1]
```diff
@@ -1626,6 +1626,28 @@ def test_pandas_univariate_dateindex_repetitions():
...
+    warning_text = (
+        "An unsupported integer index was provided and will be "
+        "converted to a range index with the same values."
+    )
+    index = pd.Index([i for i in range(3, 15)], dtype=int)
+    y = pd.DataFrame(data={0: np.arange(12)}, index=index)
+
+    # Fit the forecaster and ensure that the right warning is raised.
+    with pytest.warns(ValueWarning, match=warning_text):
+        forecaster = sarimax.SARIMAX(endog=y, k_factors=1, factor_order=1)
```
Comment (ChadFulton) [inline statsmodels/tsa/statespace/tests/test_simulate.py:1639]: `k_factors` and `factor_order` aren't arguments of `SARIMAX`.  I think you want `order`?

File: statsmodels/tsa/statespace/tests/test_simulate.py [hunk 2]
```diff
@@ -1626,6 +1626,28 @@ def test_pandas_univariate_dateindex_repetitions():
...
+        forecaster = forecaster.fit()
```
Comment (ChadFulton) [inline statsmodels/tsa/statespace/tests/test_simulate.py:1640]: Using `fit` can be problematic in tests (due to numerical instability across platforms or due to the time it takes to perform the fitting), so it's better to just use a fixed set of parameters, e.g. something like:

```python
forecaster = sarimax.SARIMAX(endog=y, order=(1, 0, 0))
forecaster = forecaster.smooth([0.5, 1.0])
```

File: statsmodels/tsa/base/tsa_model.py [hunk 3]
```diff
@@ -638,6 +638,24 @@ def _init_dates(self, dates=None, freq=None):
...
+        if (
+            isinstance(index, Index) and
+            not isinstance(index, RangeIndex) and
+            index.is_integer() and
+            index[0] != 0
+        ):
+            _index = RangeIndex(index[0], index[-1] + 1)
+            if _index.equals(index):
+                index = _index
+                _index_generated = True
```
Comment (ChadFulton) [inline statsmodels/tsa/base/tsa_model.py:651]: Due to the way the `_index_generated` flag is used in the rest of the codebase, we should keep it as `False` even though we are replacing the integer Index with a `RangeIndex` here.

**Review by ChadFulton — COMMENTED** [review]
Thanks for updating this PR! It looks good to me.  I just left a couple of easy comments.

File: statsmodels/tsa/statespace/tests/test_simulate.py (hunk 1 above)
Comment (ltsaprounis) [reply statsmodels/tsa/statespace/tests/test_simulate.py]: haha yes - I think I was using UnobservedComponents or DynamicFactor in the notebook I was testing things.
fixed now.

**Comment by ltsaprounis** [comment]
@ChadFulton - Thanks for the review!

 I have addressed the feedback now. One point that I'm not sure if you would agree with is that I now split the unsupported (but valid) indexes into 2 lists where one is the integer index (with step 1) not starting from zero to allow for testing the different warning messages and `index_generated` flags. The other option I can think of is to add this type of index into the supported ones now.

There's one failing test in the CI: [statsmodels/tsa/statespace/tests/test_dynamic_factor.py::TestDynamicFactor_ar2_errors::test_mle](https://dev.azure.com/statsmodels/statsmodels-testing/_build/results?buildId=4826&view=logs&j=3f9beef4-2a87-5b9a-d136-eeaf4b842580&t=bde20e0b-dfe9-50c5-4f8a-b07f464aecfb&l=399)
The test is failing on a tolerance issue that seems unrelated to this PR, but I'm not sure!
image

**Comment by bashtage** [comment]
That failure is pretty common on CI/Python 3.9 now.  Need to fix before release, or skip.
